rust/processor/src/worker.rs

```diff
@@ -262,119 +265,102 @@ impl Worker {
-                                Ok(()) => {},
-                                Err(e) => {
-                                    error!(
-                                        processor_name = processor_name,
-                                        stream_address = indexer_grpc_data_service_address.clone(),
-                                        error = ?e,
-                                        "[Parser] Error sending datastream response to channel."
-                                    );
-                                    panic!("[Parser] Error sending datastream response to channel.")
-                                },
-                            }
-                            info!(
-                                processor_name = processor_name,
-                                start_version = start_version,
-                                end_version = end_version,
-                                channel_size = BUFFER_SIZE - tx.capacity(),
-                                channel_recv_latency_in_secs =
-                                    last_insertion_time.elapsed().as_secs_f64(),
-                                "[Parser] Received chunk of transactions."
-                            );
-                            last_insertion_time = std::time::Instant::now();
-                        },
-                        Err(rpc_error) => {
-                            // Handle reconnection later on
-                            tracing::warn!(
-                                processor_name = processor_name,
-                                stream_address = indexer_grpc_data_service_address.clone(),
-                                error = ?rpc_error,
-                                "[Parser] Error receiving datastream response."
-                            );
-                            break;
-                        },
-                    }
-                }
-
-                // Wait for the fetched transactions to finish processing before closing the channel
-                loop {
-                    let channel_capacity = tx.capacity();
-                    info!(
-                        processor_name = processor_name,
-                        channel_size = BUFFER_SIZE - channel_capacity,
-                        "[Parser] Waiting for channel to be empty"
-                    );
-                    if channel_capacity == BUFFER_SIZE {
-                        break;
-                    }
-                    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-                }
-                info!("[Parser] The stream is ended.");
-
-                // Don't try to reconnect if we've reached ending_version
-                if let Some(ending_version) = ending_version {
-                    if next_version_to_fetch > ending_version {
+            let mut reconnection_retries = 0;
+            while let Some(current_item) = resp_stream.next().await {
```

Comment:
The reason I added reconnection logic wrapping around this while loop is because I saw logs where `[Parser] The stream is ended.` every 5 minutes, which is when the GRPC restarts. We should handle these restarts because they're expected to happen.

Reply:
yea good point. changed the logic